### ISSUE TITLE
The model resume trained from checkpoint-400 stagnated at 0% progress.

### ISSUE THREAD

**Issue by aBenNn22** [TIGER-AI-Lab/VLM2Vec#105]
Hello, when I was training your code on my own dataset and resuming training from checkpoint-400, I found that training cannot proceed and remains stuck at 0% progress. Is this related to the training still being in epoch 0? Below is my training script:
`cmd="CUDA_VISIBLE_DEVICES=0,1 torchrun --nproc_per_node=2 --master_port=2207 --max_restarts=0 train.py --lora --lora_r 8 --model_name Qwen/Qwen2-VL-2B-Instruct --bf16 --pooling eos --normalize True --temperature 0.02 --dataloader_num_workers 1 --dataset_config experiments/public/train/train_v2_m5p.yaml --run_name $EXP_NAME --output_dir $EXP_DIR --grad_cache True --per_device_train_batch_size 512 --gc_q_chunk_size 8 --gc_p_chunk_size 8 --interleave_batch_size 64 --lr_scheduler_type linear --learning_rate 5e-5 --max_steps 1000 --warmup_steps 100 --save_steps 100 --logging_steps 1 --save_safetensors True --remove_unused_columns False --resume_from $RESUME_FROM --report_to swanlab 2>&1 | tee $EXP_DIR/train.log"`
And below is the training details:
![Image](https://github.com/user-attachments/assets/eb88034a-b131-4156-9c4a-baef59706229)

**Comment by memray** [comment]
Hi @aBenNn22 ,

Can you please check if the CPU is running? 

Since we use IterableDataset and mix batches on the fly, for resuming training the data loader actually still goes through all the first 400 batches (but no modeling pass). My impression was it does take some time to resume for 400 batches but not super long (<30min, not sure). 

If it takes too long, you may consider reloading the checkpoint and starting from step 1.

Thanks,
Rui

**Comment by aBenNn22** [comment]
Thank you for replying. I'll double check if the CPU is running.
